### ISSUE TITLE
Picasso doesn't load images and never invoke callback

### ISSUE THREAD

**Issue by fraggjkee** [square/picasso#965]
Hello guys, I have a `ListView` and the following code in its adapter:

```
...

LayoutInflater layoutInflater = LayoutInflater.from(mContext);
final ViewHolder viewHolder;
if (convertView == null) {
      convertView = layoutInflater.inflate(R.layout.card_view_item_order, null);
      viewHolder = new ViewHolder();
      viewHolder.mImageView = (ImageView) convertView.findViewById(R.id.card_view_item_order_image_view);
      convertView.setTag(viewHolder);
} else {
      viewHolder = (ViewHolder) convertView.getTag();
}
Picasso.with(mContext).load(url).into(viewHolder.mImageView);

...

private static class ViewHolder {
      ImageView mImageView;
}
```

For some reason Picasso doesn't load images and also never invokes a callback if I pass it along with `ImageView` (strong reference to callback is in place), neither `onSuccess` nor `onError`. I was using Picasso in several projects before and it used to work perfectly, but for some reason it doesn't work anymore... The URL is correct and other libraries and low-level code including `InputStream` and related stuff are working well for this specific case. Any ideas? If it helps, `ImageView` has a fixed height = 100dp and MATCH_PARENT width. It is also embedded into `CardView` widget.

**Version**: 2.5.2
**Logs**:
>04-04 11:28:08.403    3736-3736/com.test D/Picasso﹕ Main        created      [R0] Request{http://damp-crag-3442.herokuapp.com/assets/android_order_item_img-b2596baa04c79bc61f403f34528edc60.png}
04-04 11:28:08.409    3736-3736/com.test D/Picasso﹕ Main        created      [R1] Request{http://damp-crag-3442.herokuapp.com/assets/android_order_item_img-b2596baa04c79bc61f403f34528edc60.png}
04-04 11:28:08.409    3736-3755/com.test D/Picasso﹕ Dispatcher  enqueued     [R0]+6ms
04-04 11:28:08.409    3736-3755/com.test D/Picasso﹕ Hunter      joined       [R1]+0ms to [R0]+6ms, [R1]+0ms
04-04 11:28:08.410    3736-3736/com.test D/Picasso﹕ Main        created      [R2] Request{http://damp-crag-3442.herokuapp.com/assets/android_order_item_img-b2596baa04c79bc61f403f34528edc60.png}
04-04 11:28:08.413    3736-3755/com.test D/Picasso﹕ Hunter      joined       [R2]+0ms to [R0]+8ms, [R1]+1ms, [R2]+0ms
04-04 11:28:08.414    3736-3773/com.test D/Picasso﹕ Hunter      executing    [R0]+11ms, [R1]+4ms, [R2]+3ms
04-04 11:28:08.921    3736-3755/com.test D/Picasso﹕ Dispatcher  retrying     [R0]+518ms, [R1]+512ms, [R2]+510ms
04-04 11:28:08.922    3736-3774/com.test D/Picasso﹕ Hunter      executing    [R0]+519ms, [R1]+513ms, [R2]+512ms

**Comment by Sserra90** [comment]
I having the same problem with picasso 2.5.2, no callback invocation, no erro on Picasso global listener, the same logs as you. The url is working fine. I get nothing, don't know how to debug this. Clearing app cache solves the problem.

**Comment by dnkoutso** [comment]
Will investigate. Thanks for the reports.

**Comment by fraggjkee** [comment]
Actually there is no need in `ListView` to reproduce this problem, a single `ImageView` is enough.

**Comment by Sserra90** [comment]
Just tested it with Ion and it's working fine with this code:

Ion.with(getActivity())
                        .load(p.photo)
                        .withBitmap()
                        .placeholder(R.drawable.default_profile)
                        .error(R.drawable.default_profile)
                        .intoImageView(vh.faceImage);

**Comment by dheerm** [comment]
I was having the same issue when I was using an older version of okhttp and not using okio.

Updating seemed to fix it.

Added the following to gradle script:

compile 'com.squareup.okio:okio:1.3.0'
compile 'com.squareup.okhttp:okhttp:2.3.0'

**Comment by kestasb** [comment]
Picasso 2.5.2 version does not display this image anymore from SD card, though works fine using 2.4.0.
![marble](https://cloud.githubusercontent.com/assets/7819283/7701871/0f50da5c-fe34-11e4-9636-8825ac299852.jpg)

I'm using .resize(thumbWidth, thumbHeight).centerCrop(), if resize() and centerCrop() is not used, image gets displayed.
Any ideas how to fix this issue?

**Comment by AndroidSuperMan** [comment]
I am having the exact issues.  When I removed this Picasso, my listview worked fine.  Of course without this Picasso, the list of pictures would take long to load.  With Picasso, the pictures never showed up on Kit Kat and up.  It works fine on 4.3 or below...  This Milestone 2.5.3 has been stuck at 69% complete.  I need this, please.... Thank you for your hard work

**Comment by neuberdesigns** [comment]
Hi guys, i have basically the same problem, but for me it happens because of fragments in the backstack.

Say i have these fragments, A,B,C,D,E,F,detail.

If i do a navigation like this: 
A > B > C > detail, it works fine and display the image.

But if i do a navigation like this:
A > B > C > detail > E > F > detail, the image its not loaded, and the callbacks are not called.

What i discover is if i already have the detail fragment into the backstack the subsequent details fragments wont work, seems to me its trying to load the image into the first detail fragment.

I will do more test but at now this is what i have.

**Comment by JakeWharton** [comment]
Closed by #1280.
